### PR TITLE
Show icons when sidebar is collapsed.

### PR DESCRIPTION
**Pull request by kounoike**
This PR may improve usage when sidebar is collapsed.

Before:
all icons are hide.
014016-root test - google chrome

After:
Show icons and badges. If you hover mouse on icon, menu is appeared.
014437-root test - google chrome

021306-root test - google chrome

And two minor feature update:

- add icons to System settings and user profile.
- add gitbucket.core.view.helpers.menuicon helper to render icon tag by name (octicon-* and fa-* are supported, default is octicon-*)

### Before submitting a pull-request to Gitbucket I have first:

- [x] read the [contribution guidelines](https://github.com/gitbucket/gitbucket/blob/master/.github/CONTRIBUTING.md)
- [x] rebased my branch over master
- [x] verified that project is compiling
- [ ] verified that tests are passing
- [x] squashed my commits as appropriate *(keep several commits if it is relevant to understand the PR)*
- [ ] [marked as closed using commit message](https://help.github.com/articles/closing-issues-via-commit-messages/) all issue ID that this PR should correct

### CODE REVIEW

**Comment by nekonenene** [comment]
Nice feature! 😄👍

File: src/main/scala/gitbucket/core/view/helpers.scala [hunk 1]
```diff
@@ -412,4 +412,30 @@ object helpers extends AvatarImageProvider with LinkConverter with RequestCache
...
+
+  /**
+    * make menu icon with octicon/font-awesome with Option[String]
+    * @param icon icon name: fa-* or octicon-*, for compatibility, it accepts name of octicon without octicon- prefix.
+    * @param defaultIcon default name for icon == None
+    * @return icon tag String
+    */
+  def menuicon(icon: Option[String], defaultIcon: String): Html = {
+    menuicon(icon.getOrElse(defaultIcon))
+  }
```
Comment (takezoe) [inline src/main/scala/gitbucket/core/view/helpers.scala]: I don't make sure whether these methods are necessary as helper. In this case, it looks enough to write `<i>` in template directly. What is your opinion?

Comment (kounoike) [inline src/main/scala/gitbucket/core/view/helpers.scala]: It uses for plugin's menuicon.
Plugin has `val repositoryMenus: Seq[(RepositoryInfo, Context) => Option[Link]]`.
and `Link` has `icon: Option[String]`. Then, plugin has icon string or None.

If plugin doesn't provide there icon, `defaultIcon` shows by this method.

**Comment by kounoike** [comment]
@takezoe I found tricky icon string in [gitbucket-commitgraph-plugin](https://github.com/yoshiyoshifujii/gitbucket-commitgraphs-plugin/blob/master/src/main/scala/Plugin.scala#L29-L37).

Please wait to modify menuicon method's replacing algorithm.

File: src/main/scala/gitbucket/core/view/helpers.scala (hunk 1 above)
Comment (takezoe) [inline src/main/scala/gitbucket/core/view/helpers.scala]: I mean both  `menuicon(icon: Option[String], defaultIcon: String)` and `def menuicon(icon:String)`.

Comment (kounoike) [inline src/main/scala/gitbucket/core/view/helpers.scala]: `def menuicon(icon: Option[String], defaultIcon: String)` is required for plugin.

Of course, we can write directory for system's icon. But it's not friendly for plugin-developer.

Comment (takezoe) [inline src/main/scala/gitbucket/core/view/helpers.scala]: Why? Plugin developers don't necesarry do any more if template is
```html
@link.icon.map { icon =>
  <i class="menu-icon octicon octicon-@icon"></i>
}.getOrElse {
  <i class="menu-icon octicon octicon-plug"></i>
}
<span>@link.label</span>
```

**Comment by takezoe** [comment]
@kounoike I tried to fix some points of this pull request.
https://github.com/gitbucket/gitbucket/compare/kounoike-pr-sidebar-mini?expand=1

I hope you will adopt this for your request. Could you check it and tell your opinion?

**Comment by kounoike** [comment]
@takezoe Sorry for the late response.
It looks good to me, but I wonder to three `-<div class="container body">` changes. What is it?

**Comment by takezoe** [comment]
@kounoike `<div class="container body">` was nested.

They make tiny unnecessary margins on both sides of the body content as:

broken_layout

**Comment by takezoe** [comment]
Merged in 4e299b2ad51e12ef690db5c3da0616b58c9fd002

**Pull request closed without merging**